cql2pgjson-cli/src/main/java/com/indexdata/cql2pgcli/CQL2PGCLIMain.java

```diff
@@ -0,0 +1,122 @@
+package com.indexdata.cql2pgcli;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.cli.*;
+import org.json.JSONException;
+import org.z3950.zing.cql.cql2pgjson.CQL2PgJSON;
+import org.z3950.zing.cql.cql2pgjson.FieldException;
+import org.z3950.zing.cql.cql2pgjson.QueryValidationException;
+import org.z3950.zing.cql.cql2pgjson.SchemaException;
+import org.z3950.zing.cql.cql2pgjson.SqlSelect;
+import org.json.JSONObject;
+
+public class CQL2PGCLIMain { 
+  
+  public static void main( String[] args ) {
+    Options options = new Options();
+    
+    Option database = Option.builder("d")
+        .hasArg()
+        .required(true)
+        .desc("Postgres database name")
+        .build();
+    
+    Option field = Option.builder("f")
+        .hasArg()
+        .required(true)
+        .desc("Postgres field name")
+        .build();
+    
+    Option schema = Option.builder("s")
+        .hasArg()
+        .desc("Path to JSON schema file")
+        .build();
+    
+    Option dbschema = Option.builder("m")
+        .hasArg()
+        .desc("A JSON string or a pathname to a JSON file to describe the db/schema map")
+        .build();
+    
+    options.addOption(schema);
+    options.addOption(database);
+    options.addOption(field);
+    options.addOption(dbschema);
+    CommandLineParser parser = new DefaultParser();
+    CQL2PgJSON cql2pgJson = null;
+    try {
+      CommandLine line = parser.parse(options, args);
+      String fullDbName = line.getOptionValue("d") + "." + line.getOptionValue("f");
+      if(!line.hasOption("m")) {
+        if(line.hasOption("s")) {       
+          String schemaText = readFile(line.getOptionValue("s"), Charset.forName("UTF-8"));   
+          cql2pgJson = new CQL2PgJSON(fullDbName, schemaText);
+        } else {
+          cql2pgJson = new CQL2PgJSON(fullDbName); //No schemas
+        }
+      } else {
+        Map<String, String> fieldSchemaMap = parseDatabaseSchemaString(line.getOptionValue("m"));
+        cql2pgJson = new CQL2PgJSON(fieldSchemaMap);        
+      }
+      List<String> cliArgs = line.getArgList();  
+      String cql = cliArgs.get(0);
```

Comment:
What happens with the other arguments?

Reply:
The CQL string is the only argument it takes.